beacon-chain/sync/initial-sync/service.go

```diff
@@ -244,6 +244,18 @@ func (s *InitialSync) run(chainHeadResponses map[peer.ID]*pb.ChainHeadResponse)
 		return chainHeadResponses[peers[i]].CanonicalSlot > chainHeadResponses[peers[j]].CanonicalSlot
 	})
 
+	// Retry for our 3 best peers
+	totalNumTries := 3
+	peersToRetry := 3
+	bestPeers := make([]peer.ID, totalNumTries*peersToRetry)
+	for i := range bestPeers {
+		indexToAdd := i / totalNumTries
+		bestPeers[i] = peers[indexToAdd]
```

Comment:
is this right?
`indexToAdd` will be `0/3, 1/3, 2/3` to start with...

Reply:
yeah its an int, so it will all be still 0. 
the peer list should look like this. 
[0,0,0,1,1,1,2,2,2,3,4,5,6..]
We retry the 3 best peers an extra 2 times

Reply:
Gotcha, thanks

Reply:
Why only retry the 3 best peers? Why not retry all of them?

Reply:
This seems like a griefing factor, if i can connect to you with three nodes and say that I am the best, then there is a chance I can cause you to never sync.